### PR TITLE
solc dependency updated to 0.7.4

### PR DESCRIPTION
**Pull request by Aniket-Engg**

### CODE REVIEW

File: libs/remix-debug/test/debugger.js [hunk 1]
```diff
@@ -258,7 +258,7 @@ function testDebugging (debugManager) {
 
   tape('traceManager.decodeLocalsAt', async (t) => {
     t.plan(1)
-    const tested = JSON.parse('{"proposalNames":{"value":[{"value":"0x48656C6C6F20576F726C64210000000000000000000000000000000000000000","type":"bytes32"}],"length":"0x1","type":"bytes32[]","cursor":1,"hasNext":false},"p":{"value":"45","type":"uint256"},"addressLocal":{"value":"0x4B0897B0513FDC7C541B6D9D7E929C4E5364D2DB","type":"address"},"i":{"value":"2","type":"uint256"},"proposalsLocals":{"value":[{"value":{"name":{"value":"0x48656C6C6F20576F726C64210000000000000000000000000000000000000000","type":"bytes32"},"voteCount":{"value":"0","type":"uint256"}},"type":"struct Ballot.Proposal"}],"length":"0x1","type":"struct Ballot.Proposal[]"}}')
+    const tested = JSON.parse('{"proposalNames":{"value":[{"value":"0x48656C6C6F20576F726C64210000000000000000000000000000000000000000","type":"bytes32"}],"length":"0x1","type":"bytes32[]","cursor":1,"hasNext":false}}')
```
Comment (yann300) [inline libs/remix-debug/test/debugger.js:261]: that means that at `330`, all the locals are not defined. But we should find the correct vmtrace step where all locals are defined and use it in the test.

File: libs/remix-debug/test/decoder/contracts/byteStorage.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ module.exports = {
     bytes32 stab32 = hex"324324423432543543AB";
     enum1  enumDec = enum1.e240;
     string str1 = 'short';
-    string str12 = 'шеллы';
```
Comment (yann300) [inline libs/remix-debug/test/decoder/contracts/byteStorage.js:47]: why renaming this?

File: libs/remix-debug/test/decoder/localsTests/misc.js [hunk 3]
```diff
@@ -46,9 +46,9 @@ module.exports = function (st, vm, privateKey, contractBytecode, compilationResu
             st.equals(locals['__bytes9'].value, '0x99156744AF00000000')
             st.equals(locals['__bytes13'].value, '0x99123423425300000000000000')
             st.equals(locals['__bytes16'].value, '0x99AFAD23432400000000000000000000')
-            st.equals(locals['__bytes24'].value, '0x99AFAD234324000000000000000000000000000000000000')
-            st.equals(locals['__bytes32'].value, '0x9999ABD41799ABD4170000000000000000000000000000000000000000000000')
-            st.equals(Object.keys(locals).length, 16)
+            // st.equals(locals['__bytes24'].value, '0x99AFAD234324000000000000000000000000000000000000')
```
Comment (yann300) [inline libs/remix-debug/test/decoder/localsTests/misc.js:49]: I think that is the same as above, as the byte code changed, we should update step.

File: libs/remix-debug/test/sourceLocationTracker.js [hunk 4]
```diff
@@ -63,12 +63,12 @@ tape('SourceLocationTracker', function (t) {
         st.equal(map.start, 35)
 
         map = await sourceLocationTracker.getSourceLocationFromVMTraceIndex('0x0d3a18d64dfe4f927832ab58d6451cecc4e517c5', 45, output.contracts)
-        st.equal(map.file, -1)
```
Comment (yann300) [inline libs/remix-debug/test/sourceLocationTracker.js:66]: ah I understand the question earlier now.
so this is not that the src map has changed, but the bytecode and thus the vm trace has changed

File: libs/remix-debug/test/decoder/contracts/byteStorage.js (hunk 2 above)
Comment (Aniket-Engg) [reply libs/remix-debug/test/decoder/contracts/byteStorage.js:47]: compiler was throwing error for this

File: libs/remix-debug/test/sourceLocationTracker.js (hunk 4 above)
Comment (Aniket-Engg) [reply libs/remix-debug/test/sourceLocationTracker.js:66]: ok, got it.

File: libs/remix-debug/test/decoder/contracts/byteStorage.js (hunk 2 above)
Comment (yann300) [reply libs/remix-debug/test/decoder/contracts/byteStorage.js:47]: it doesn't accept such characters anymore?

Comment (Aniket-Engg) [reply libs/remix-debug/test/decoder/contracts/byteStorage.js:47]: It throws this error: `ParserError: Invalid character in string. string str12 = 'шеллы'; ^^`

File: libs/remix-debug/test/decoder/localsTests/misc.js (hunk 3 above)
Comment (Aniket-Engg) [reply libs/remix-debug/test/decoder/localsTests/misc.js]: updated

File: libs/remix-debug/test/decoder/contracts/byteStorage.js (hunk 2 above)
Comment (Aniket-Engg) [reply libs/remix-debug/test/decoder/contracts/byteStorage.js:47]: This is fixed by writing it as: `string str12 = unicode'шеллы';`

File: libs/remix-debug/test/debugger.js (hunk 1 above)
Comment (Aniket-Engg) [reply libs/remix-debug/test/debugger.js]: updated

File: libs/remix-debug/test/sourceLocationTracker.js [hunk 5]
```diff
@@ -63,12 +63,12 @@ tape('SourceLocationTracker', function (t) {
...
+        st.equal(map.file, 1)
```
Comment (yann300) [inline libs/remix-debug/test/sourceLocationTracker.js:66]: file was `-1` previously

File: libs/remix-debug/test/sourceLocationTracker.js [hunk 6]
```diff
@@ -63,12 +63,12 @@ tape('SourceLocationTracker', function (t) {
...
 
         map = await sourceLocationTracker.getValidSourceLocationFromVMTraceIndex('0x0d3a18d64dfe4f927832ab58d6451cecc4e517c5', 45, output.contracts)
-        st.equal(map.file, 0)
-        st.equal(map.start, 303)
-        st.equal(map.length, 448)
+        st.equal(map.file, 1)
```
Comment (yann300) [inline libs/remix-debug/test/sourceLocationTracker.js:69]: there's more than one file in `output.contracts` ?

Comment (Aniket-Engg) [reply libs/remix-debug/test/sourceLocationTracker.js:69]: no

**Review by yann300 — APPROVED** [review]

**Pull request merged**